providers/src/airflow/providers/amazon/aws/transfers/s3_to_dynamodb.py

```diff
@@ -237,10 +237,13 @@ def _load_into_existing_table(self) -> str:
                 total_items += page.get("Count", 0)
                 dynamodb_hook.write_batch_data(items=page["Items"])
             self.log.info("Number of items loaded: %s", total_items)
-        finally:
-            self.log.info("Delete tmp DynamoDB table %s", self.tmp_table_name)
-            client.delete_table(TableName=self.tmp_table_name)
-            return dynamodb_hook.get_conn().Table(self.dynamodb_table_name).table_arn
+        except Exception as e:
+            # Does not raise errors to keep previous behavior.
+            self.log.warning("Error while loading data from temp table, deleting temp table. Error: %s", e)
+
+        self.log.info("Delete tmp DynamoDB table %s", self.tmp_table_name)
+        client.delete_table(TableName=self.tmp_table_name)
```

Comment:
don't we want to keep it in a finally clause?

Reply:
Yes, maybe it's better to keep them in finally clause, in case of exception not handled by the try clause.